201-front-door-premium-vm-private-link/modules/front-door.bicep

```diff
@@ -0,0 +1,167 @@
+@allowed([
+  'Standard_AzureFrontDoor'
+  'Premium_AzureFrontDoor'
+])
+param skuName string
+
+@description('The protocol that should be used when connecting from Front Door to the origin.')
+@allowed([
+  'HttpOnly'
+  'HttpsOnly'
+  'MatchRequest'
+])
+param originForwardingProtocol string = 'HttpsOnly'
+
+@description('If you are using Private Link to connect to the origin, this should specify the resource ID of the Private Link resource (e.g. an App Service application, Azure Storage account, etc). If you are not using Private Link then this should be empty.')
+param privateEndpointResourceId string = ''
+
+@description('If you are using Private Link to connect to the origin, this should specify the resource type of the Private Link resource. The allowed value will depend on the specific Private Link resource type you are using. If you are not using Private Link then this should be empty.')
+param privateLinkResourceType string = ''
+
+@description('If you are using Private Link to connect to the origin, this should specify the location of the Private Link resource. If you are not using Private Link then this should be empty.')
+param privateEndpointLocation string = ''
+
+// When connecting to Private Link origins, we need to assemble the privateLinkOriginDetails object with various pieces of data.
+var isPrivateLinkOrigin = (privateEndpointResourceId != '')
+var privateLinkOriginDetails = {
+  privateLink: {
+    id: privateEndpointResourceId
+  }
+  groupId: (privateLinkResourceType != '') ? privateLinkResourceType : null
+  privateLinkLocation: privateEndpointLocation
+  requestMessage: 'Please approve this connection.'
+}
+
+var profileName = 'MyFrontDoor'
+var originGroupName = 'MyOriginGroup'
+var originName = 'MyOrigin'
+var routeName = 'MyRoute'
+
+resource profile 'Microsoft.Cdn/profiles@2020-09-01' = {
+  name: profileName
+  location: 'global'
+  sku: {
+    name: skuName
+  }
+}
+
+resource endpoint 'Microsoft.Cdn/profiles/afdEndpoints@2020-09-01' = {
+  name: '${profile.name}/${endpointName}'
+  location: 'global'
+  properties: {
+    originResponseTimeoutSeconds: 240
+    enabledState: 'Enabled'
+  }
+}
+
+resource originGroup 'Microsoft.Cdn/profiles/originGroups@2020-09-01' = {
+  name: '${profile.name}/${originGroupName}'
+  properties: {
+    loadBalancingSettings: {
+      sampleSize: 4
+      successfulSamplesRequired: 3
+    }
+    healthProbeSettings: {
+      probePath: '/'
+      probeRequestType: 'HEAD'
+      probeProtocol: 'Http'
+      probeIntervalInSeconds: 100
+    }
+  }
+}
+
+resource origin 'Microsoft.Cdn/profiles/originGroups/origins@2020-09-01' = {
+  name: '${originGroup.name}/${originName}'
+  properties: {
+    hostName: originHostName
+    httpPort: 80
+    httpsPort: 443
+    originHostHeader: endpoint.properties.hostName // This probably isn't the right host header for most scenarios - you will need to look at what makes sense for your solution.

```

Comment:
this seems an odd comment - can we be more prescriptive?  (or opinionated?)

Reply:
This is a fairly contrived sample (since the purpose of the sample isn't to configure a VM fully, but rather to show how Front Door works with Private Link service and VMs at a network topology level). I've rejigged the way this works to clarify that the way this is set up is specific to the way the sample has been designed.

This point is also mentioned in the README.md file too.

Reply:
Should the comment be "this isn't the right host header for ....  see the readme/docs for more detail?"

The "probably" wording makes me ask "well is it or isn't it?"

Reply:
I've updated the comment - the 'probably' is in an outdated version.

The new comment is:

> For the purposes of this sample, the 'Host' header value sent from Front Door to the origin must be the Front Door endpoint's hostname. In most real-world scenarios, you will need to set this to the originHostName value or a custom domain name, as described here: https://docs.microsoft.com/azure/frontdoor/standard-premium/concept-origin#hostheader